core/domain/activity_jobs_one_off.py

```diff
@@ -471,3 +478,129 @@ def map(snapshot_model):
+            },
+            'SkillSnapshotMetadataModel': {
+                'parent_model': skill_models.SkillModel,
+                'commit_log_model': skill_models.SkillCommitLogEntryModel,
+                'id_string_format': 'skill-%s-%s'
+            },
+            'StorySnapshotMetadataModel': {
+                'parent_model': story_models.StoryModel,
+                'commit_log_model': story_models.StoryCommitLogEntryModel,
+                'id_string_format': 'story-%s-%s'
+            },
+            'TopicSnapshotMetadataModel': {
+                'parent_model': topic_models.TopicModel,
+                'commit_log_model': topic_models.TopicCommitLogEntryModel,
+                'id_string_format': 'topic-%s-%s'
+            },
+            'SubtopicPageSnapshotMetadataModel': {
+                'parent_model': subtopic_models.SubtopicPageModel,
+                'commit_log_model': (
+                    subtopic_models.SubtopicPageCommitLogEntryModel),
+                'id_string_format': 'subtopicpage-%s-%s'
+            },
+            'TopicRightsSnapshotMetadataModel': {
+                'parent_model': topic_models.TopicRightsModel,
+                'commit_log_model': topic_models.TopicCommitLogEntryModel,
+                'id_string_format': 'rights-%s-%s'
+            },
+            'CollectionRightsSnapshotMetadataModel': {
+                'parent_model': collection_models.CollectionRightsModel,
+                'commit_log_model': (
+                    collection_models.CollectionCommitLogEntryModel),
+                'id_string_format': 'rights-%s-%s'
+            },
+            'ExplorationRightsSnapshotMetadataModel': {
+                'parent_model': exp_models.ExplorationRightsModel,
+                'commit_log_model': exp_models.ExplorationCommitLogEntryModel,
+                'id_string_format': 'rights-%s-%s'
+            },
+        }
+
+        val = model_dict[class_name]
+        commit_log_id = val['id_string_format'] % (model_id, version)
+        parent_model = val['parent_model'].get_by_id(model_id)
+        commit_log_model = val['commit_log_model'].get_by_id(commit_log_id)
+        if class_name in rights_model:
+            if snapshot_model.commit_type in ['create', 'delete']:
+                missing_commit_log_msg = 'COMMIT LOGS SHOULD NOT EXIST'
+
+        if commit_log_model is None:
+            yield (
+                '%s - %s' % (missing_commit_log_msg, class_name),
+                snapshot_model.id)
+        else:
+            yield ('FOUND COMMIT LOGS - %s' % class_name, snapshot_model.id)
+
+        if parent_model is None:
+            yield ('MISSING PARENT MODEL - %s' % class_name, snapshot_model.id)
+        else:
+            yield ('FOUND PARENT MODEL - %s' % class_name, snapshot_model.id)
+
+    @staticmethod
+    def reduce(key, values):
+        """Implements the reduce function for this job."""
+        if key.startswith('MISSING'):
```

Comment:
Perhaps define a value FAILURE_PREFIX = 'VALIDATION FAILURE' and, when constructing the error message, you can do it as '%s %s - %s' % (FAILURE_PREFIX, error_message, ...). Then, in reduce(), refer to FAILURE_PREFIX.

In general, don't define constants as raw strings multiple times. Typos may result in them not matching up, so always use named constants if you want to reuse a constant multiple times.

Reply:
Done.